### ISSUE TITLE
Add size preferences for images at filename selection

### ISSUE THREAD

**Issue by msdobrescu** [arsenetar/dupeguru#728]
Hello, I use dupeguru under Linux, I build it and I run it.

My goal is to select nice wallpapers from my large collection, I take directory by directory and compare the candidate wallpapers against the already selected ones.
I use picture mode, 80% hardness, match pictures of different dimensions.
I set the selected wallpapers directory (all are there per categories) and one or few candidates directory.
What I've tried was to get the biggest (in resolution) wallpapers I can. There are many similar and duplicates of different sizes.
After doing this, I have come to the conclusion that, although I compare them myself in some cases, I need to have for the found duplicates sets the biggest one as reference.
Then, what does it mean biggest?
Generally, two attributes come in my mind:
- the file size itself in bytes, especially when the files have the same resolution, as I prefer the better quality
- the resolution, I need the biggest ones for my 2K monitor, usually the width over the height, but this may be also height over the width sometimes.
So for now, I end up switching a lot the references in the results with some of their duplicates and this is counterproductive.

Is it possible to have a reference preference unrelated to the directories, but to images attributes as above? Also, to have the possibility to set the preference to resolution over file size or vice-versa?

Thank you.

**Comment by glubsy** [comment]
You can use the "Re-prioritize" feature to sort by file size. A resolution re-prioritize feature could be considered maybe.

**Comment by msdobrescu** [comment]
Indeed, I didn't know this feature, but I'd prefer to be a default behaviour setting rather than doing it after each comparison. Maybe prioritization is good to be available at the settings level.

**Comment by atescula** [comment]
Dupeguru is amazing. Saved my "day" after harddisk failure some years ago. Following a complicated recovery process for two weeks I got 2.5 million picures (initial photo collection was around 80000).

Then it was only dupeguru (photo mode) to help me get rid of visually same pictures but binary different.

I suggest using xnviewmp and / or digikam (I am on linux platform) to help sort pictures by separating in folders / collections according to your need and then make use of dupeguru.

Use of exiftool command line utility is also advised (on windows maybe there is exiftool GUI) but using exiftool in linux terminal is more efficient.
Este cel mai bun utilitar in acest scop, cel putin pe linux.

Greetings

**Comment by msdobrescu** [comment]
Thanks, but no, dupeguru only. This is not about my photos, it's about wallpapers downloaded from the Internet, so no exif, no tags.

**Comment by msdobrescu** [comment]
Working as suggested, I have found that close to what I need, but I still find myself working a lot on switching between reference and selection. I would need to be able to have a prioritization like this:
- keep the highest resolution
- keep the biggest file in bytes
- keep specific folders when the match is 100%

Regarding the above, I rely on dimensions column or deltas, but I'd prefer to see both at once.

What do you think?
